framework/src/problems/FEProblemBase.C

```diff
@@ -3130,8 +3133,9 @@ FEProblemBase::addMaterialHelper(std::vector<MaterialWarehouse *> warehouses,
 
     bool discrete = !material->getParam<bool>("compute");
 
-    // If the object is boundary restricted do not create the neighbor and face objects
-    if (material->boundaryRestricted())
+    // If the object is boundary restricted or if it is a functor material we do not create the
+    // neighbor and face objects
+    if (material->boundaryRestricted() || dynamic_cast<FunctorMaterial *>(material.get()))
```

Comment:
> i think we want to avoid the else statement here too

I don't understand what you mean. If the material is a `FunctorMaterial` then we don't make it into the `else` branch

> is there a similar optimization with variables?

The main reason that I have a distinct `FunctorMaterial` class is so that I can ensure that we don't do triplicate addition of `FunctorMaterials` like we do for regular `Materials`. If you have the triplicate addition, then you can't do sanity checks for whether only one material provides a functor material property on a given subdomain. We do not do triplicate addition of variables like we do for regular materials

Reply:
oh was misreading this if statement